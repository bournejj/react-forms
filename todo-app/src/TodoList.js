import React, {useState} from "react";
import {v4 as uuid} from 'uuid'
import Todo from "./Todo.js";
import NewTodoForm from "./NewTodoForm.js";

const TodoList = () => {
    const INITIAL_STATE = [
        {id: "", todo: ""}
    ]
    const [items, setItems] = useState(INITIAL_STATE);
    
        const addItem = (todo) => {
            setItems(items => [...items, {id: uuid(), todo}])
    
        }

       const deleteTodo = id => {
           setItems(items => {
               const copy = [...items];
               const item = id
               const filteredArray =  copy.filter(function(el) { return el.id !== item; });
               console.log(filteredArray)
               return filteredArray;

           })

       }
            






    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm addItem={addItem}/>
            <div>
                {items.map(({id, todo}) => <Todo deleteBox={deleteTodo} id ={id} todo={todo} />)}
                
            </div>
            




        </div>
    )
}

export default TodoList;
