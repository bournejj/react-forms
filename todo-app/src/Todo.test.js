import React from "react";
import {render, fireEvent} from "@testing-library/react";
import TodoList from "./TodoList.js";
import { query } from "express";

it("renders without crashing", function () {
    render(<TodoList/>);
})

it("matches snapshot", function () {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
    
})
it("it should add new item", function () {
    const {queryByText, getByLabelText} = render(<TodoList/>);
    const input = getByLabelText('todo');
    const btn = getByText('Add Todo');
    expect(queryByText('todo: wash car')).not.toBeInTheDocument();
    // fireEvent.change(input, {target: {value: 'wash car'}});
    // fireEvent.click(btn);
    // expect(queryByText('todo: wash car')).toBeInTheDocument();
    
    
})

