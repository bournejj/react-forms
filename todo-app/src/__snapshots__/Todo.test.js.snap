// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should add new item 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="todo"
      >
        todo
      </label>
      <input
        name="todo"
        type="text"
        value=""
      />
      <p />
      <button>
        Add Todo
      </button>
    </form>
    <div>
      <ul>
        <li />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="todo"
      >
        todo
      </label>
      <input
        name="todo"
        type="text"
        value=""
      />
      <p />
      <button>
        Add Todo
      </button>
    </form>
    <div>
      <ul>
        <li />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
